components/mime.rst

```diff
@@ -0,0 +1,541 @@
+    $twig = new Environment($templateLoader);
+
+    $renderer = new BodyRenderer($twig);
+    // this updates the $email object contents with the result of rendering
+    // the template defined earlier with the given context
+    $renderer->render($email);
+
+The last step is to create the Twig template used to render the contents:
+
+.. code-block:: html+twig
+
+    <h1>Welcome {{ username }}!</h1>
+
+    <p>You signed up to our site using the following email:</p>
+    <p><code>{{ email.to }}</code></p>
+
+    <p><a href="{{ url('...') }}">Click here to activate your account</a></p>
+
+The Twig template has access to any of the parameters passed in the ``context``
+method of the ``TemplatedEmail`` class and also to a special variable called
+``email`` which gives access to any of the email message properties.
+
+If you don't define the text content of the message, the ``BodyRenderer()``
+class generates it automatically converting the HTML contents into text. If you
+prefer to define the text content yourself, use the ``text()`` method explained
+in the previous sections or the ``textTemplate()`` method provided by the
+``TemplatedEmail`` class::
+
+    use Symfony\Bridge\Twig\Mime\TemplatedEmail;
+
+    $email = (new TemplatedEmail())
+        ->from('fabien@symfony.com')
+        ->fo('foo@example.com')
+        // ...
+
+        ->textTemplate('messages/user/signup.txt.twig')
+        ->htmlTemplate('messages/user/signup.html.twig')
+
+        // this method defines the parameters (name => value) passed to templates
+        ->context([
+            // same as before...
+        ])
+    ;
+
+Inlining CSS Styles in Emails with Twig
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Designing the HTML contents of an email is very different from designing a
+normal HTML page. For starters, most email clients only support a subset of all
+CSS features. In addition, popular email clients such as Gmail don't support
+defining styles inside ``<style> ... </style>`` sections and you must **inline
+all the CSS styles**.
+
+CSS inlining means that every HTML tag must define a ``style`` attribute with
+all its CSS styles. This not only increases the email byte size significantly
+but also makes it impossible to manage for complex emails. That's why Twig
+provides a ``CssInlinerExtension`` that automates everything for you. First,
+install the Twig extension in your application:
+
+.. code-block:: terminal
+
+    $ composer require twig/cssinliner-extension
+
+Now, enable the extension (this is done automatically in Symfony applications)::
+
+    // ...
+    use Twig\CssInliner\CssInlinerExtension;
+
+    $templateLoader = new FilesystemLoader(__DIR__.'/templates');
+    $twig = new Environment($templateLoader);
+    $twig->addExtension(new CssInlinerExtension());
+
+Finally, wrap the entire template contents with the ``inline_css`` filter
+(to do so, it's better to use the `{% filter %} Twig tag`_ instead of the
+traditional ``...|filter`` notation):
```

Comment:
I would remove the content of the parenthesis, not sure it adds something useful.